src/modules/launcher/Wox.Infrastructure/Image/ImageLoader.cs

```diff
@@ -144,92 +143,94 @@ private static ImageResult LoadInternal(string path, bool generateThumbnailsFrom
+                        if (extension == ".png" || generateThumbnailsFromFiles)
                         {
-                            image = LoadFullImage(path);
+                            /* Although the documentation for GetImage on MSDN indicates that
+                            * if a thumbnail is available it will return one, this has proved to not
+                            * be the case in many situations while testing.
+                            * - Solution: explicitly pass the ThumbnailOnly flag
+                            */
+                            image = WindowsThumbnailProvider.GetThumbnail(path, Constant.ThumbnailSize, Constant.ThumbnailSize, ThumbnailOptions.ThumbnailOnly);
                         }
                         else
                         {
-                            // PowerToys Run internal images are png, so we make this exception
-                            if (extension == ".png" || generateThumbnailsFromFiles)
-                            {
-                                /* Although the documentation for GetImage on MSDN indicates that
-                                * if a thumbnail is available it will return one, this has proved to not
-                                * be the case in many situations while testing.
-                                * - Solution: explicitly pass the ThumbnailOnly flag
-                                */
-                                image = WindowsThumbnailProvider.GetThumbnail(path, Constant.ThumbnailSize, Constant.ThumbnailSize, ThumbnailOptions.ThumbnailOnly);
-                            }
-                            else
-                            {
-                                image = WindowsThumbnailProvider.GetThumbnail(path, Constant.ThumbnailSize, Constant.ThumbnailSize, ThumbnailOptions.IconOnly);
-                            }
+                            image = WindowsThumbnailProvider.GetThumbnail(path, Constant.ThumbnailSize, Constant.ThumbnailSize, ThumbnailOptions.IconOnly);
                         }
                     }
-                    else if (!generateThumbnailsFromFiles || (extension == ".pdf" && WindowsThumbnailProvider.DoesPdfUseAcrobatAsProvider()))
-                    {
-                        // The PDF thumbnail provider from Adobe Reader and Acrobat Pro lets crash PT Run with an Dispatcher exception. (https://github.com/microsoft/PowerToys/issues/18166)
-                        // To not run into the crash, we only request the icon of PDF files if the PDF thumbnail handler is set to Adobe Reader/Acrobat Pro.
-                        // Also don't get thumbnail if the GenerateThumbnailsFromFiles option is off.
-                        type = ImageType.File;
-                        image = WindowsThumbnailProvider.GetThumbnail(path, Constant.ThumbnailSize, Constant.ThumbnailSize, ThumbnailOptions.IconOnly);
-                    }
```

Comment:
I don't think we should remove it. On my test system Adobe's provider shows always black (and broken) images in explorer.

We should keep it and maybe hide it behind a setting to enable it if Adobe thumbnails makes trouble.

Reply:
This would be covered by "Generate thumbnails from files" disabled option as a fallback, right?

Reply:
But then you have no thumbnails for all types.

Reply:
True. Although, I'm not too fond of the idea of keep adding these exceptional cases. I guess we can keep covering this one as AdobeReader is quite popular and widely used software and hopefully won't end up adding a bunch of these

Reply:
reverted